pkg/storage/bucket/s3/config.go

```diff
@@ -44,6 +44,7 @@ var (
 	errUnsupportedStorageClass     = fmt.Errorf("unsupported S3 storage class (supported values: %s)", strings.Join(supportedStorageClasses, ", "))
 	errInvalidSSEContext           = errors.New("invalid S3 SSE encryption context")
 	errInvalidEndpointPrefix       = errors.New("the endpoint must not prefixed with the bucket name")
+	errInvalidSTSPrefix            = errors.New("sts-endpoint must be a valid url")
```

Comment:
`errInvalidSTSPrefix` -> `errInvalidSTSEndpoint`

I think prefix refers to an actual prefix of the endpoint in `errInvalidEndpointPrefix`. Here the entire sts endpoint is invalid

Reply:
fixed.